src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1268,6 +1286,26 @@ bool entitiesQuery
 
                 cer->contextElement.contextAttributeVector.push_back(caP);
             }
+        }        
+
+        /* All the attributes existing in the request but not found in the response are added with 'found' set to false */
+        for (unsigned int ix = 0; ix < attrL.size(); ++ix)
+        {
+          bool found = false;
+          std::string attrName = attrL.get(ix);
+          for (unsigned int jx = 0; jx < cer->contextElement.contextAttributeVector.size(); ++jx)
+          {
+            if (attrName == cer->contextElement.contextAttributeVector.get(jx)->name)
+            {
+              found = true;
+              break;
+            }
+          }
+          if (!found)
+          {
+            ContextAttribute* caP = new ContextAttribute(attrName, "", "", false);
+            cer->contextElement.contextAttributeVector.push_back(caP);
+          }
```

Comment:
Not sure if this works... note that I create & push if I *don't find* anything (not if I find).

Maybe if you could show me the complete code snipped it would be more clear to me...